client/beefy/src/tests.rs

```diff
@@ -528,85 +511,77 @@ fn finalize_block_and_wait_for_beefy(
 	if expected_beefy.is_empty() {
 		// run for quarter second then verify no new best beefy block available
 		let timeout = Some(Duration::from_millis(250));
-		streams_empty_after_timeout(best_blocks, &net, runtime, timeout);
-		streams_empty_after_timeout(versioned_finality_proof, &net, runtime, None);
+		streams_empty_after_timeout(best_blocks, &net, timeout).await;
+		streams_empty_after_timeout(versioned_finality_proof, &net, None).await;
 	} else {
 		// run until expected beefy blocks are received
-		wait_for_best_beefy_blocks(best_blocks, &net, runtime, expected_beefy);
-		wait_for_beefy_signed_commitments(versioned_finality_proof, &net, runtime, expected_beefy);
+		wait_for_best_beefy_blocks(best_blocks, &net, expected_beefy).await;
+		wait_for_beefy_signed_commitments(versioned_finality_proof, &net, expected_beefy).await;
 	}
 }
 
-#[test]
-fn beefy_finalizing_blocks() {
+#[tokio::test]
+async fn beefy_finalizing_blocks() {
 	sp_tracing::try_init_simple();
 
-	let mut runtime = Runtime::new().unwrap();
 	let peers = [BeefyKeyring::Alice, BeefyKeyring::Bob];
 	let validator_set = ValidatorSet::new(make_beefy_ids(&peers), 0).unwrap();
 	let session_len = 10;
 	let min_block_delta = 4;
 
-	let mut net = BeefyTestNet::new(runtime.handle().clone(), 2);
+	let mut net = BeefyTestNet::new(2);
 
 	let api = Arc::new(two_validators::TestApi {});
 	let beefy_peers = peers.iter().enumerate().map(|(id, key)| (id, key, api.clone())).collect();
-	runtime.spawn(initialize_beefy(&mut net, beefy_peers, min_block_delta));
+	tokio::spawn(initialize_beefy(&mut net, beefy_peers, min_block_delta));
```

Comment:
What happens with this task when the test ends, is it also killed because test is marked as `#[tokio::test]`. I mean when multiple `#[tokio::test]`s are run concurrently does tokio have some magic under the hood to know which `tokio::spawn()` belongs to which test?

Old version of the code had clear task lifetime -> the lifetime of `runtime`; with this code, I'm not sure. If you know the answer, please also add a code-comment for others like me 😄

Reply:
`#[tokio::test]` is equivalent to something like `Runtime::new().unwrap().block_on(async {...})`, so every test gets its own runtime that is dropped at the end. I little more accurate descriptions is in the [docs](https://docs.rs/tokio/latest/tokio/attr.test.html).

@acatangiu I'm afraid there are too many `tokio::spawn()` across the PR to add comments (I counted 44 =)

Reply:
thank you! [tokio::test](https://docs.rs/tokio/latest/tokio/attr.test.html) and [tokio::spawn](https://docs.rs/tokio/latest/tokio/fn.spawn.html) documentation clearly supports what you are saying, sorry for not researching it before asking. It's fine to skip the comments.